custom_components/google_home/__init__.py

```diff
@@ -41,15 +40,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.data.setdefault(DOMAIN, {})
         _LOGGER.info(STARTUP_MESSAGE)
 
-    username = entry.data.get(CONF_USERNAME)
-    password = entry.data.get(CONF_PASSWORD)
     android_id = entry.data.get(CONF_ANDROID_ID)
+    master_token = entry.data.get(CONF_MASTER_TOKEN)
 
     session = async_get_clientsession(hass, verify_ssl=False)
 
     zeroconf_instance = await zeroconf.async_get_instance(hass)
     glocaltokens_client = GlocaltokensApiClient(
-        hass, username, password, session, android_id, zeroconf_instance
+        hass,
+        session,
+        master_token=master_token,
+        android_id=android_id,
+        zeroconf_instance=zeroconf_instance,
     )
```

Comment:
I really think we should have a fallback to `username` and `password` if `master_token` is not there.
We should provide "all we have" and let `glocaltokens` package decide what should be used. 

We can safely keep `username`, `password` and add `master_token` here, because looking at how `glocaltokens` client is working now it will not use `username` and `password` if `master_token` was provided when initializing `GLocalAuthenticationTokens` client

Reply:
It would also be very nice to make all arguments `kwargs`